### PR TITLE
Convert autoplay article from WebFu.

### PR DESCRIPTION
**Pull request by jpmedley**

### CODE REVIEW

File: site/_data/i18n/tags.yml [hunk 1]
```diff
@@ -11,6 +11,9 @@ accessibility:
 amp:
   en: AMP
 
+autoplay:
+  en: autoplay
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/_data/i18n/tags.yml:15]: It's literally the name of an attribute.

**Comment by jpmedley** [comment]
Why are these in a single bullet? What's the connection between them that lead you to make them a single idea?

**Comment by beaufortfrancois** [comment]
For testing purposes, developers can force Chrome to block autoplay with these two flags together. Something like that works for me:

> You can also decide to make sure autoplay is never allowed by disabling MEI, and whether sites with the highest overall MEI get playback autoplay by default for new users. This can be done with flags: `chrome.exe --disable-features=PreloadMediaEngagementData, MediaEngagementBypassAutoplayPolicies`.

File: site/_data/i18n/tags.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/_data/i18n/tags.yml:15]: Which instance is visible to readers? This or the one in supportedTags?

**Comment by jpmedley** [comment]
@beaufortfrancois I've updated the last bullet.

File: site/en/blog/autoplay/index.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+tags:
+  - autoplay
+  - media
+---
+
+{% Aside %}The Autoplay Policy launched in Chrome 66 for audio and video
+elements and is effectively blocking roughly half of unwanted media autoplays in
+Chrome. For the Web Audio API, the autoplay policy launched in Chrome 71. This
+affects web games, some WebRTC applications, and other web pages using audio
+features. More details can be found in the [Web Audio API](#webaudio) section
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:21]: The "[Web Audio API]" link does not work.

image

File: site/en/blog/autoplay/index.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+ad blockers, and reduce data consumption on expensive and/or constrained
+networks. These changes are intended to give greater control of playback to
+users and to benefit publishers with legitimate use cases.
+
+Chrome's autoplay policies are simple:
+
+- Muted autoplay is always allowed.
+- Autoplay with sound is allowed if:
+    - The user has interacted with the domain (click, tap, etc.).
+    - On desktop, the user's [Media Engagement Index](#media-engagement-index)
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:65]: This internal link does not work. Please update with `#media-engagement-index-mei` or use an hardcoded one (better)

File: site/en/blog/autoplay/index.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      threshold has been crossed, meaning the user has previously played video
+      with sound.
+    - The user has [added the site to their home screen] on mobile or
+      [installed the PWA](installed-the-pwa) on desktop.
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:69]: This link does not work. Please update with https://web.dev/progressive-web-apps/

File: site/en/blog/autoplay/index.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+Chrome's autoplay policies are simple:
+
+- Muted autoplay is always allowed.
+- Autoplay with sound is allowed if:
+    - The user has interacted with the domain (click, tap, etc.).
+    - On desktop, the user's [Media Engagement Index](#media-engagement-index)
+      threshold has been crossed, meaning the user has previously played video
+      with sound.
+    - The user has [added the site to their home screen] on mobile or
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:68]: This link does not work. Please update with https://web.dev/customize-install/

File: site/en/blog/autoplay/index.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+      [installed the PWA](installed-the-pwa) on desktop.
+- Top frames can [delegate autoplay permission](#iframe-delegation) to their iframes to
+  allow autoplay with sound.
+
+
+### Media Engagement Index (MEI)
+
+The MEI measures an individual's propensity to consume media on a site.
+Chrome's [current approach] is a ratio of visits to significant media playback
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:77]: This link does not exist anymore. You can use `Chrome's approach is ...` instead.

File: site/en/blog/autoplay/index.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+From that, Chrome calculates a media engagement score, which is highest on sites
+where media is played on a regular basis. When it is high enough, media is
+allowed to autoplay on desktop only.
+
+A user's MEI is available at the `about://media-engagement` internal page.
+
+<figure class="w-figure">
+  {% Img src="image/sQ51XsLqKMgSQMCZjIN0B7hlBO02/x0XKHYcLKjmDzjfxQxFg.png", alt="Screenshot of about://media-engagement internal page.", width="800", height="280" %}
+  <figcaption class="w-figcaption">
+    Screenshot of the <code>chrome://media-engagement</code> internal page.
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:94]:
```suggestion
     Screenshot of the <code>about://media-engagement</code> internal page in Chrome.

File: site/en/blog/autoplay/index.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+  </figcaption>
+</figure>
+
+### Developer switches
+
+As a developer, you may want to change Chrome autoplay policy behavior locally
+to test your website for different levels of user engagement.
+
+- You can disable the autoplay policy entirely by using a [command line flag]:
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:103]: `[command line flag]` link does not work. 

image

File: site/en/blog/autoplay/index.md [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+### Iframe delegation
+
+A [permissions policy] allows developers to selectively enable and disable
+browser features and APIs. Once an origin has received autoplay
+permission, it can delegate that permission to cross-origin iframes with the
+[permissions policy for autoplay]. Note that autoplay is allowed by default on
+same-origin iframes.
+
+```html
+&lt;!-- Autoplay is allowed. -->
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:123]: Please fix html formatting issues.
image

File: site/en/blog/autoplay/index.md [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+**Example 4:** `MyMovieReviewBlog.com` embeds an iframe with a movie trailer to
+go with a review. Users interacted with the domain to get to the blog, so
+autoplay is allowed. However, the blog needs to explicitly delegate that
+privilege to the iframe in order for the content to autoplay.
+
+### Chrome enterprise policies
+
+It is possible to change the autoplay behavior with Chrome enterprise policies
+for use cases such as kiosks or unattended systems. Check out the [Policy
+List](policy-list) help page to learn how to set the autoplay related enterprise
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:163]: Use https://chromeenterprise.google/policies/ for policy-list URL

File: site/en/blog/autoplay/index.md [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+policies:
+
+- The `[AutoplayAllowed](autoplayallowed)` policy controls whether autoplay is allowed or not.
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:166]: Please fix formatting issues:
image

File: site/en/blog/autoplay/index.md [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+
+Here's the one thing to remember: Don't ever assume a video will play, and
+don't show a pause button when the video is not actually playing. It is so
+important that I'm going to write it one more time below for those who simply
+skim through that post.
+
+{% Aside 'gotchas' %}Don't assume a video will play, and don't show a pause
+button when the video is not actually playing.{% endAside %}
+
+You should always look at the [Promise](promise) returned by the play function
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:182]: Promise link does not work. Use https://web.dev/promises/

File: site/en/blog/autoplay/index.md [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+to see if it was [rejected]:
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:183]: "rejected" link does not work. Use https://developers.google.com/web/updates/2017/06/play-request-was-interrupted

File: site/en/blog/autoplay/index.md [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+
+### Web Audio
+
+The [Web Audio API](web-audio) has been covered by autoplay since Chrome 71.
+There are a few things to know about it. First, it is good practice to wait for
+a user interaction before starting audio playback so that users is aware of
+something happening. Think of a "play" button or "on/off" switch for instance.
+You can also add an "unmute" button depending on the flow of the app.
+
+{% Aside %}If an `AudioContext` is created before the document receives a user
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:225]: Fix formatting issue
image

File: site/en/blog/autoplay/index.md [hunk 15]
```diff
@@ -0,0 +1,290 @@
...
+gesture, it will be created in the "suspended" state, and you will need to call
+`resume()` after the user gesture.{% endAside %}
+
+If you create your `AudioContext` on page load, you'll have to call `resume()`
+at some time after the user interacted with the page (e.g., after a user clickes
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:230]: clickes? I believe it should be clicks or clicked no?

File: site/en/blog/autoplay/index.md [hunk 16]
```diff
@@ -0,0 +1,290 @@
...
+```js
+document.querySelector('button').addEventListener('click', function() {
+  var context = new AudioContext();
+  // Setup all nodes
+  // ...
+});
+```
+
+To detect whether the browser requires a user interaction to play audio, check
+`AudioContext.state` after you've created it. If your media is allowed to play,
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:262]: There's no "media". I believe it should "If you are allowed to play," as it's more generic. WebAudio is different than regular audio/video elements.

File: site/en/blog/autoplay/index.md [hunk 17]
```diff
@@ -0,0 +1,290 @@
...
+it should immediately switch to `running`. Otherwise it will be `suspended`. If
+you listen to the `statechange` event, you can detect changes asynchronously.
+
+To see an example, check out the small [Pull Request](pull-request) that fixes
+Web Audio playback for these autoplay policy rules for [https://airhorner.com].
+
+{% Aside %}You can find asummary of [Chrome's autoplay
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:269]: Fix formatting issues.

image

**Review by beaufortfrancois — CHANGES_REQUESTED** [review]
You may want to stage the article locally to spot formatting issues easily. Thank you!

File: site/en/blog/autoplay/index.md (hunk 16 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:262]: Used "if playing is allowed."

**Comment by jpmedley** [comment]
@beaufortfrancois PTAL. Thanks for your help with the links.

**Comment by beaufortfrancois** [comment]
@jpmedley https://deploy-preview-801--developer-chrome-com.netlify.app/blog/autoplay is not updated with your latest changes which makes it harder for me to review. Can you fix this?

**Comment by beaufortfrancois** [comment]
> @jpmedley https://deploy-preview-801--developer-chrome-com.netlify.app/blog/autoplay is not updated with your latest changes which makes it harder for me to review. Can you fix this?

(gentle ping)

**Comment by jpmedley** [comment]
@beaufortfrancois Sorry. I have been out of the office. I wish GH had OOO replies. 

I thought the test deploy was supposed to happen automatically. @robdodson, please help.

**Comment by robdodson** [comment]
[This comment](https://github.com/GoogleChrome/developer.chrome.com/pull/801#issuecomment-844311030) says that it was built from commit 1d2ea9e, which seems like the latest one from Joe:

image

Can you clear your cache and try again. If you're still not seeing the correct updates then I would recommend building the site locally to preview. If Netlify is out of sync then the only solution I can think of is to recreate the PR but that seems like it would be annoying given how close y'all are to landing this.

**Comment by beaufortfrancois** [comment]
@jpmedley (gentle ping)

File: site/_data/i18n/tags.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/_data/i18n/tags.yml:15]: HI. Which version is visible?

File: site/en/blog/autoplay/index.md (hunk 17 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: Looks fixed now.

File: site/en/blog/autoplay/index.md (hunk 16 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 15 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 14 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 11 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:169]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 10 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:166]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 9 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:124]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 8 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:104]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 6 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: Done.

File: site/en/blog/autoplay/index.md (hunk 4 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:70]: Done.

File: site/en/blog/autoplay/index.md (hunk 5 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 3 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:66]: Fixed.

File: site/en/blog/autoplay/index.md (hunk 2 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: I think it's fixed.

File: site/en/blog/autoplay/index.md (hunk 12 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md:187]: I think it's fixed.

File: site/en/blog/autoplay/index.md (hunk 13 above)
Comment (jpmedley) [reply site/en/blog/autoplay/index.md]: I think it's fixed.

**Comment by jpmedley** [comment]
PTAL

**Comment by beaufortfrancois** [comment]
@jpmedley As you can see in the screenshot below, this PR fails to "lint, test and build". 
image

I suggest you do as @robdodson said:
> If Netlify is out of sync then the only solution I can think of is to recreate the PR but that seems like it would be annoying given how close y'all are to landing this.

**Comment by robdodson** [comment]
I think something is broken with that check, I've seen it stall like this before. Unfortunately GitHub doesn't let you manually run it.

If Joe has approved the PR I am happy to merge it

**Comment by beaufortfrancois** [comment]
@robdodson  I'm the one to approve it actually and having the live preview would help

**Comment by robdodson** [comment]
Yeah in that case I think you'll either need to recreate the PR or test it locally. I'd recommend testing it locally if you have the site handy.

**Comment by beaufortfrancois** [comment]
I've cloned this Git repo, checked out the PR locally, and noticed links formatting is wrong.
@jpmedley Check out https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links to see the difference between inline-style and reference-style links. Then, resolve conflicts file `site/_data/authorsData.json` and I'll have a last look hopefully.

**Comment by jpmedley** [comment]
@beaufortfrancois I think I have it this time. PTAL

**Comment by jpmedley** [comment]
@robdodson Percy is complaining about a page that's not part of this PR.

File: site/en/blog/autoplay/index.md [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+    video.muted = false;
+  });
+</script>
+```
+
+### Web Audio
+
+The [Web Audio API][web audio api] has been covered by autoplay since Chrome 71.
+There are a few things to know about it. First, it is good practice to wait for
+a user interaction before starting audio playback so that users is aware of
```
Comment (beaufortfrancois) [inline site/en/blog/autoplay/index.md:226]: Nit: `are aware of`

**Review by beaufortfrancois — APPROVED** [review]
LGTM with nit

**Review by robdodson — APPROVED** [review]
Approving to unlock pullapprove

**Pull request merged**